guides/advanced_guides/datatypes.md

```diff
@@ -0,0 +1,135 @@
+
+Based on their type, however, the fields will be handled and used in different ways by MeiliSearch. **The type affects how a field is used for search results**.
+
+Types:
+
+- [String](/guides/advanced_guides/datatypes.md#string)
+- [Numeric types](/guides/advanced_guides/datatypes.md#numeric-types): `integer`, `float`
+- [Boolean](/guides/advanced_guides/datatypes.md#boolean)
+- [Array](/guides/advanced_guides/datatypes.md#array)
+- [Object](/guides/advanced_guides/datatypes.md#object)
+- [`null` type](/guides/advanced_guides/datatypes.md#null-type)
+
+### String
+
+String is the primary type for indexing data in MeiliSearch. It enables to create the content in which to search. Strings are processed as detailed below.
+
+> String tokenization is the process of splitting a string into a list of individual terms that are called tokens.
+
+A string is passed to a tokenizer and is then broken into separate string tokens. A token is a **word**.
+
+- For Latin-based languages, the words are separated by **space**.
+- For Kanji characters, the words are separated by **character**.
+
+For Latin-based languages, there are two kinds of **space separators**:
+
+- **Soft spaces** (distance: 1): whitespaces, quotes, `'-' | '_' | '\'' | ':' | '/' | '\\' | '@'`
+- **Hard spaces** (distance: 8): `'.' | ';' | ',' | '!' | '?' | '(' | ')'`
+
+Distance plays an essential role in in determining whether documents are relevant since [one of the ranking rules is the **proximity** rule](/guides/main_concepts/relevancy.md). The proximity rule sorts the results by increasing distance between matched query terms. Then, two words separated by a soft space are closer and thus considered **more relevant** than two words separated by a hard space.
+
+After the tokenizing process, each word is indexed and stored in the global dictionary of the corresponding index.
+
+#### Examples
+
+To demonstrate how a string is split by space, let's say you have the following string as an input:
+
+```
+"Bruce Willis,Vin Diesel"
+```
+
+In the example above, the distance between `Bruce` and `Willis` is equal to **1**. The distance between `Vin` and `Diesel` is equal to **1** too.
+But, the distance between `Bruce` and `Vin` is equal to **8**. The same goes for `Bruce` and `Diesel`, or `Vin` and `Willis`, or also `Willis` and `Diesel`.
+
+Let's see another example. Given two documents:
+
+```json
+[
+  {
+    "movie_id": "001",
+    "description": "Bruce.Willis"
+  },
+  {
+    "movie_id": "002",
+    "description": "Bruce super Willis"
+  }
+]
+```
+
+When making a query on `Bruce Willis`, `002` will be the first document returned and `001` will be the second one.
+This will happen because the proximity distance between `Bruce` and `Willis` is equal to **7** in the document `002` whereas the distance between `Bruce` and `Willis` is equal to **8** in the document `001` since the full stop is a hard space.
```

Comment:
The distance is on a word basis not characters.

```suggestion
This will happen because the proximity distance between `Bruce` and `Willis` is equal to **2** in the document `002` whereas the distance between `Bruce` and `Willis` is equal to **8** in the document `001` since the full stop is a hard space.
```

Reply:
Thanks!